File: providers/uniapi.py
"""
UniAPI 余额查询适配器
"""
from .base import ProviderSpec, SimpleHTTPProvider


def _check(data):
    if not data.get('success'):
        return "API 返回 success=false"
    return None


def _extract(data):
    balance = (data.get('data') or {}).get('balance')
    if balance is None:
        raise ValueError("无法从响应中解析 balance 字段")
    return float(balance)


class UniAPIProvider(SimpleHTTPProvider):
    """UniAPI 服务商适配器"""

    SPEC = ProviderSpec(
        name="UniAPI",
        url="https://api.uniapi.io/v1/billing/usage",
        params={"unit": "usd"},
        check=_check,
        extract=_extract,
    )

File: internal/provider/uniapi.go
package provider

import "errors"

// Implementation note.
var uniapiSpec = Spec{
	Key:         "uniapi",
	Name:        "UniAPI",
	DefaultType: "credits",
	URL:         "https://api.uniapi.io/v1/billing/usage",
	Params:      map[string]string{"unit": "usd"},
	Check: func(data map[string]any) error {
		if !truthy(data["success"]) {
			return errors.New("API returned success=false")
		}
		return nil
	},
	Extract: func(data map[string]any) (float64, error) {
		balance, ok := Num(Dig(data, "data", "balance"))
		if !ok {
			return 0, errors.New("Could not parse balance field")
		}
		return balance, nil
	},
}

func init() { RegisterSpec(uniapiSpec) }
